lib/modules/varnish/poll.c

```diff
@@ -65,27 +65,38 @@ static const struct {
 size_t varnish_fetch_metrics(varnish_module_t *mod, const char *output) {
   size_t n_metrics = 0;
   char *found;
+  char *output_start;
   char metric_name[256];
   struct timeval clock;
 
   gettimeofday(&clock, NULL);
+  output_start = output;
   for (size_t i = 0; metrics[i].name; i++) {
-    if ((found = strstr(output, metrics[i].varnish_name))
-    && ((found - VARNISH_SMA_TRANSIENT_LEN < output)
-        || strncmp(found - VARNISH_SMA_TRANSIENT_LEN, VARNISH_SMA_TRANSIENT,
-            VARNISH_SMA_TRANSIENT_LEN))) {
-      char *endval;
-      char *val = found + strlen(metrics[i].name);
-      uint64_t value = strtoull(val, &endval, 10);
-      if (val == endval)
-        continue;
-      if (mod->instance)
-        sprintf(metric_name, "%s.%s", metrics[i].name, mod->instance);
-      else
-        sprintf(metric_name, "%s", metrics[i].name);
-      n_metrics += monikor_metric_push(mod->mon, monikor_metric_integer(
-        metric_name, &clock, value, metrics[i].flags));
+    while (found = strstr(output, metrics[i].varnish_name)) {
```

Comment:
I'd do `for (found = output; (found = strstr(found, metrics[i].varnish_name)); found += strlen(metrics[i].varnish_name))` instead
Please note the extra parenthesis around `(found = ...` which were missing on the while (thus always evaluating to true because an assignment is always true)